### PR TITLE
minor refactoring

### PR DESCRIPTION
**Pull request by brualan**
What is the purpose of this change? What does it change?
--------------------------------------------------------

This is the first step in solving the problems found by static analyzers. The ultimate goal: reduce the amount of code, increase readability, improve performance

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have added tests for all changes in this PR
- [x] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changlog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

These are the first commits I want to make. Not sure how it will be better: gradually do the integration of the PR or complete the work and merge one large PR. I use `golangci-lint run --enable-all` to find problem areas of code

### CODE REVIEW

File: internal/options/options.go [hunk 1]
```diff
@@ -22,10 +22,8 @@ func Register(ns string, cfg interface{}) {
 
 // List returns a list of all registered options (using Register()).
 func List() (list []Help) {
-	list = make([]Help, 0, len(opts))
-	for _, opt := range opts {
-		list = append(list, opt)
-	}
+	list = make([]Help, len(opts))
+	copy(list, opts)
```
Comment (fd0) [inline internal/options/options.go]: Oh, wow, how did I not see that!

File: internal/cache/cache.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ type Cache struct {
 }
 
 const dirMode = 0700
-const fileMode = 0600
+const fileMode = 0644
```
Comment (fd0) [inline internal/cache/cache.go]: Good find!

**Review by fd0 — APPROVED** [review]
Awesome, thank you very much for taking the time to clean this up! When I have a bit more spare time I plan to replace Hound with golang-ci, so we get these kinds of checks for all PRs.

Also, it was very helpful that you split the changes by type, so I could easily review them!

**Comment by brualan** [comment]
Hi, I’m pretty busy lately. As soon as I have the opportunity, I will continue the work. Please merge this PR, so that it does not hang a dead weight and the time spent on it was not wasted

**Comment by fd0** [comment]
Will do, thanks for your work!

**Pull request merged**